Core/Src/main.c

```diff
@@ -87,14 +107,49 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
+  MX_TIM2_Init();
   /* USER CODE BEGIN 2 */
 
+  // Start the PWM timer
+  HAL_TIM_PWM_Start(&htim1, TIM_CHANNEL_1);
+  HAL_TIM_PWM_Start(&htim2, TIM_CHANNEL_1);
+
+  // Pull CS Pin high to ensure no communication happens
+  HAL_GPIO_WritePin(SPI1_CSB_GPIO_Port, SPI1_CSB_Pin, GPIO_PIN_SET);
+
   /* USER CODE END 2 */
 
   /* Infinite loop */
   /* USER CODE BEGIN WHILE */
+
   while (1)
   {
+	  // Pull CS Pin low to start SPI communication
+	  HAL_GPIO_WritePin(SPI1_CSB_GPIO_Port, SPI1_CSB_Pin, GPIO_PIN_RESET);
+
+	  // Transmit SPI data
+	  HAL_SPI_Transmit(&hspi1, SPI_TRANSMIT, SPI_TRANSMIT_SIZE, SPI_TIMEOUT);
+
+	  // Receive SPI data
+	  HAL_SPI_Receive(&hspi1, SPI_Recieve, SPI_RECIEVE_SIZE, SPI_TIMEOUT);
```

Comment:
Oh, good to know that either using the transmit_and_recieve function or the functions separately works!